src/nodedumper.cc

```diff
@@ -39,35 +39,33 @@ void NodeDumper::handleIndent(const State &state)
 	including braces and indentation.
 	All children are assumed to be cached already.
  */
-std::string NodeDumper::dumpChildBlock(const AbstractNode &node)
+void NodeDumper::dumpChildBlock(const AbstractNode &node, std::stringstream &dump) const
 {
-	std::stringstream dump;
-	if (!this->visitedchildren[node.index()].empty()) {
+	const auto &it = this->visitedchildren.find(node.index());
+
+	if (it == this->visitedchildren.end() || it->second.empty()) {
+		dump << ";";
+	} else {
 		dump << " {\n";
-		const auto &chstr = dumpChildren(node);
-		if (!chstr.empty()) dump << chstr << "\n";
+		dumpChildren(node, dump);
```

Comment:
I think I'll merge this as is, but this constructs makes me think that we should manage stream operators better.
A quickfix could be to return the stream reference from `dumpChildren()`, making this construct look more natural, e.g:
```
  dump << " {\n" 
       << dumpChildren(node, dump)
       << this->currindent
       << "}";
```